### PR TITLE
version for review

### PR DESCRIPTION
**Pull request by FedotkoV**
..but not good enough

### CODE REVIEW

File: .idea/deploymentTargetDropDown.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shadwork) [inline .idea/deploymentTargetDropDown.xml]: remove this file from repo and add it to gitignore

File: .idea/gradle.xml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="GradleMigrationSettings" migrationVersion="1" />
```
Comment (shadwork) [inline .idea/gradle.xml]: remove this file from repo and add it to gitignore

File: .idea/misc.xml [hunk 3]
```diff
@@ -3,8 +3,12 @@
   <component name="DesignSurface">
```
Comment (shadwork) [inline .idea/misc.xml]: remove this file from repo and add it to gitignore

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import android.os.Bundle
+import android.text.Editable
+import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+
+class LogInActivity : AppCompatActivity() {
+    lateinit var userName: EditText
+    lateinit var userPassword: EditText
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: use View Binding https://habr.com/ru/post/467295/

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    lateinit var logInButton: Button
+    private val passwordLength = 6
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: should be configurable as constant

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    /**
+     * checks user`s input if user name is correct start to check him password
+     */
+    private fun checkUser() {
+        val userNameToText = userName.text
+
+        if (userNameToText.isEmpty()) {
+            Toast.makeText(this, "name cannot be empty",
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: Should be SnackBar

Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: Move text into android resources

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+     */
+    private fun checkPassword() {
+        val userPasswordString = userPassword.text
+
+        when  {
+            (userPasswordString.isEmpty()) ->
+                Toast.makeText(this, "password cannot be empty",
+                    Toast.LENGTH_SHORT).show()
+            ' ' in userPasswordString -> {
+                Toast.makeText(this, "password cannot contain space sign",
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: all text should be in android resources

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                inputMethodManager.hideSoftInputFromWindow(logInButton.windowToken, 0)
+
+                Toast.makeText(this, "You logged in",
+                    Toast.LENGTH_LONG).show()
+            }
+        }
+    }
+
+    private fun hasSmallLetter(userPassword: Editable): Boolean {
+        for (i in 'a'..'z') {
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: should be checked view Pattern Matcher

File: app/src/main/java/com/getstarted/lesson1/MainActivity.kt [hunk 9]
```diff
@@ -1,11 +1,94 @@
...
         setContentView(R.layout.activity_main)
+
+        image = findViewById(R.id.image_view)
+
+        if (savedInstanceState != null) {
+            imageIndex = savedInstanceState.getInt("imageIndex")
+        }
+
+        //  starts the thread
+        Handler(Looper.getMainLooper()).postDelayed({
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/MainActivity.kt]: Handler should cleared on pause

File: app/src/main/java/com/getstarted/lesson1/MainActivity.kt [hunk 10]
```diff
@@ -1,11 +1,94 @@
 package com.getstarted.lesson1
```
Comment (shadwork) [inline app/src/main/java/com/getstarted/lesson1/MainActivity.kt:1]: Good solution, but using coroutines instead of thread please

**Review by shadwork — CHANGES_REQUESTED** [review]
Норм, но давай перепишем анимацию на корутины

File: .idea/deploymentTargetDropDown.xml (hunk 1 above)
Comment (FedotkoV) [reply .idea/deploymentTargetDropDown.xml]: done

File: .idea/gradle.xml (hunk 2 above)
Comment (FedotkoV) [reply .idea/gradle.xml]: done

File: .idea/misc.xml (hunk 3 above)
Comment (FedotkoV) [reply .idea/misc.xml]: done

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt (hunk 4 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt (hunk 5 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt (hunk 6 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt (hunk 7 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/LogInActivity.kt (hunk 8 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/LogInActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/MainActivity.kt (hunk 9 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/MainActivity.kt]: done

File: app/src/main/java/com/getstarted/lesson1/MainActivity.kt (hunk 10 above)
Comment (FedotkoV) [reply app/src/main/java/com/getstarted/lesson1/MainActivity.kt:1]: done

**Pull request merged**